sdk/appconfiguration/app-configuration/test/internal/helpers.spec.ts

```diff
@@ -130,6 +139,40 @@ describe("helper methods", () => {
     });
   });
 
+  describe("serializeAsConfigurationSettingParam", () => {
+    [[], `[]`, "Hello World"].forEach((value) => {
+      it(`serializer doesn't throw on ${value} as feature flag value`, () => {
+        const featureFlag: ConfigurationSetting<FeatureFlagValue> = {
+          contentType: featureFlagContentType,
+          key: "key",
+          isReadOnly: false,
+          value: { conditions: { clientFilters: [] }, enabled: true }
+        };
+        featureFlag.value = value as any;
+        assert.equal(
+          serializeAsConfigurationSettingParam(featureFlag),
+          featureFlag,
+          "setting was modified"
+        );
+      });
+
+      it(`serializer doesn't throw on ${value} as secret reference value`, () => {
+        const setting: ConfigurationSetting<SecretReferenceValue> = {
+          contentType: secretReferenceContentType,
+          key: "key",
+          isReadOnly: false,
+          value: { secretId: "abc" }
+        };
+        setting.value = value as any;
+        assert.equal(
```

Comment:
Doesn't this need to be a deepEqual?

Reply:
Also, I would think that the 'key' would be transformed to be the featureFlag format?

Reply:
> Doesn't this need to be a deepEqual?

Updated

> Also, I would think that the 'key' would be transformed to be the featureFlag format?

Not in this case because the setting is not recognized as a feature flag.